nomad/state/state_store.go

```diff
@@ -242,7 +242,7 @@ func (s *StateStore) SnapshotMinIndex(ctx context.Context, index uint64) (*State
 		// Get the states current index
 		snapshotIndex, err := s.LatestIndex()
 		if err != nil {
-			return nil, fmt.Errorf("failed to determine state store's index: %v", err)
+			return nil, fmt.Errorf("failed to determine state store's index: %w", err)
```

Comment:
I'm :+1: for wrapping all the errors we can, but FWIW this is one of the many cases where we can't ever fail b/c the inner call to [`txn.Get`](https://github.com/hashicorp/go-memdb/blob/main/txn.go#L744) only returns an error if we give it the wrong table or index. (I'd be willing to make an argument we should panic in the state store helpers around those methods, but that's certainly out of scope for this PR.)